test/blackbox-tests/test-cases/pkg/alias-pkg-install.t

```diff
@@ -0,0 +1,49 @@
+This test verifies the @pkg-install alias fetch and build the project dependencies
+without building the project itself.
+
+  $ . ./helpers.sh
+
+Create a fake package which echoes information to stdout when build:
+  $ make_lockdir
+  $ cat > dune.lock/foo.pkg << EOF
+  > (version 0.0.1)
+  > (build
+  >  (run echo "Build package foo"))
+  > (install
+  >  (run echo "Install package foo"))
+  > EOF
+
+Create a project using the fake library as a dependency:
+  $ cat > dune-project << EOF
+  > (lang dune 3.16)
+  > (package
+  >  (name bar)
+  >  (allow_empty)
+  >  (depends foo))
+  > EOF
+
+Create a rule to show that this rule is not called with `@pkg-install` as `bar`
+is not build when calling the alias. If called, it would output the content of
+the `bar.ml` file:
+  $ cat > dune << EOF
+  > (executable
+  >  (name bar))
+  > (rule
+  >  (target bar.ml)
+  >  (action
+  >   (progn
+  >    (with-stdout-to %{target} (echo "let _ = 42"))
+  >    (system "cat %{target}"))))
+  > EOF
```

Comment:
You could simplify this to a `bar.ml` that just contains `let () = print_endline "Running bar"` and then instead of `dune build` use `dune exec`.

(I'm assuming that `dune exec` would show all the `echo` that `dune build` would when building the dependencies; otherwise `dune build ./bar.exe` followed by `dune exec ./bar.exe` avoids all the complex processing for `bar.ml`)

Reply:
This issue with this is that we are more interested in the cat than the content of `bar.ml`.  With such a configuration, it will only show that `dune exec` works as expected, not that `bar.exe` is not built with `@pkg-install`. Unless I misunderstood your suggestion :sweat_smile: